File: test-goldens/java/full_module.java
import java.util.ArrayList;
import java.util.List;

import javax.annotation.Nullable;

public interface UserRepository {
    @Nullable
    User findById(String id);

    List<User> findAll();
}

/**
 * In-memory implementation of UserRepository.
 */
public class InMemoryUserRepository implements UserRepository {
    private final List<User> users;

    public InMemoryUserRepository() {
        this.users = new ArrayList<>();
    }

    @Override
    @Nullable
    public User findById(String id) {
        return this.users.stream()
            .filter(u -> u.getId().equals(id))
            .findFirst()
            .orElse(null);
    }

    @Override
    public List<User> findAll() {
        return new ArrayList<>(this.users);
    }
}

File: test-goldens/kotlin/full_module.kt
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.MutableList

internal interface UserRepository {
    fun findById(id: String): User?

    fun findAll(): List<User>
}

/**
 * In-memory implementation of UserRepository.
 */
internal class InMemoryUserRepository : UserRepository {
    private val users: MutableList<User>

    internal override fun findById(id: String): User? {
        return users.firstOrNull { it.id == id }
    }

    internal override fun findAll(): List<User> {
        return ArrayList<User>(users)
    }
}
